src/coreComponents/constitutive/solid/DelftEgg.hpp

```diff
@@ -0,0 +1,618 @@
+  arrayView1d< real64 const > const m_virginCompressionIndex;
+
+  /// A reference to the ArrayView holding the slope of the critical state line for each element.
+  arrayView1d< real64 const > const m_cslSlope;
+
+  /// A reference to the ArrayView holding the shape parameter for each element.
+  arrayView1d< real64 const > const m_shapeParameter;
+
+  /// A reference to the ArrayView holding the new preconsolidation pressure for each integration point
+  arrayView2d< real64 > const m_newPreConsolidationPressure;
+
+  /// A reference to the ArrayView holding the old preconsolidation presure for each integration point
+  arrayView2d< real64 > const m_oldPreConsolidationPressure;
+
+};
+
+
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void DelftEggUpdates::evaluateYield( real64 const p,
+                                     real64 const q,
+                                     real64 const pc,
+                                     real64 const M,
+                                     real64 const alpha,
+                                     real64 const Cc,
+                                     real64 const Cr,
+                                     real64 const bulkModulus,
+                                     real64 const mu,
+                                     real64 & f,
+                                     real64 & df_dp,
+                                     real64 & df_dq,
+                                     real64 & df_dpc,
+                                     real64 & df_dp_dve,
+                                     real64 & df_dq_dse ) const
+{
+  real64 const c = alpha/(alpha+1.)*pc;
+  real64 a = alpha;
+  real64 pa = pc;
+  real64 factor = 1.0;
+
+  if( p >= c ) // Use MCC
+  {
+    a = 1.0;
+    factor = 2.*alpha/ (alpha+1.);
+    pa = factor * pc;
+  }
+  real64 alphaTerm = 2. * a*a*a / (a+1.);
+  df_dp = -alphaTerm * pa + 2. * a * a * p;
+  df_dq = 2. * q /(M*M);
+  df_dpc = 2. * a*a*(a-1.) /(a+1.) * pc - alphaTerm * p * factor;
+  real64 dpc_dve = -1./(Cc-Cr) * pc;
+  df_dp_dve = 2. * a * a * bulkModulus + alphaTerm * dpc_dve * factor;
+  df_dq_dse = 2. /(M*M) * 3. * mu;
+
+  f = q*q/(M*M)- a*a*p *(2.*a/(a+1.)*pa-p)+a*a*(a-1.)/(a+1.)* pc*pc;
+
+}
+
+
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void DelftEggUpdates::smallStrainUpdate( localIndex const k,
+                                         localIndex const q,
+                                         real64 const ( &strainIncrement )[6],
+                                         real64 ( & stress )[6],
+                                         real64 ( & stiffness )[6][6] ) const
+{
+
+  // Rename variables for easier implementation
+
+  real64 const oldPc  = m_oldPreConsolidationPressure[k][q];   //pre-consolidation pressure
+  real64 const mu     = m_shearModulus[k];
+  real64 const bulkModulus     = m_bulkModulus[k];
+
+  real64 const M      = m_cslSlope[k];
+  real64 const Cr     = m_recompressionIndex[k];
```

Comment:
How does the recompression index relate to the elastic bulk modulus?  Is this only used in the hardening law?

Reply:
Yes it is used only in the hardening law for Delft Egg model because we are using linear elasticity. I kept it as a separate input instead of deriving it from bulk modulus to remain consistent with MCC. Would you suggest removing the recompression index as input?